===================================================================
RCS file: /repository/php4/ext/iconv/iconv.c,v
retrieving revision 1.38
retrieving revision 1.39
diff -p --unified=3 -r1.38 -r1.39
--- php4/ext/iconv/iconv.c	2002/04/28 01:37:54	1.38
+++ php4/ext/iconv/iconv.c	2002/06/28 07:12:32	1.39
@@ -236,7 +236,7 @@ static int php_iconv_string(const char *
 		out_buf = tmp_buf;
 		out_p = tmp_buf;
 		out_p += out_size;
-		out_left = in_len;
+		out_left = bsz;
 		result = iconv(cd, (char **)&in_p, &in_left, &out_p, &out_left);
 		out_size += bsz - out_left;
 	}
